src/modules/ens/sagas.ts

```diff
@@ -71,78 +72,86 @@ function* handleConnectWallet() {
+      const nodehash = namehash(subdomain)
+      const ensContract = ENS__factory.connect(ENS_ADDRESS, signer)
+
+      const resolverAddress: string = yield call(() => ensContract.resolver(nodehash))
+
+      if (resolverAddress.toString() === ethers.constants.AddressZero) {
+        yield put(
+          fetchENSSuccess({
+            name,
+            address,
+            subdomain,
+            resolver: ethers.constants.AddressZero,
+            content: ethers.constants.AddressZero
+          })
+        )
+        return
+      }
 
-    if (resolverAddress.toString() === ethers.constants.AddressZero) {
-      yield put(
-        fetchENSSuccess({
-          name,
-          address,
-          subdomain,
-          resolver: ethers.constants.AddressZero,
-          content: ethers.constants.AddressZero
-        })
-      )
-      return
-    }
+      const resolverContract = ENSResolver__factory.connect(resolverAddress, signer)
 
-    const resolverContract = ENSResolver__factory.connect(resolverAddress, signer)
-    const ipfsHash: string = yield call(() => ipfs.uploadRedirectionFile(land))
-    const landHash = contentHash.fromIpfs(ipfsHash)
+      const [x, y] = getCenter(getSelection(land))
 
-    const currentContent: string = yield call(() => resolverContract.contenthash(nodehash))
-    if (currentContent === ethers.constants.AddressZero) {
-      yield put(
-        fetchENSSuccess({
-          address,
-          name,
-          subdomain,
-          resolver: resolverAddress.toString(),
-          content: ethers.constants.AddressZero,
-          ipfsHash
-        })
+      const { ipfsHash, contentHash }: LandHashes = yield call(
+        [builderClient, builderClient.uploadLandRedirectionFile],
+        { x, y },
+        getCurrentLocale().locale
       )
-      return
-    }
 
-    if (`0x${landHash}` === currentContent) {
+      const currentContent: string = yield call(() => resolverContract.contenthash(nodehash))
+      if (currentContent === ethers.constants.AddressZero) {
+        yield put(
+          fetchENSSuccess({
+            address,
+            name,
+            subdomain,
+            resolver: resolverAddress.toString(),
+            content: ethers.constants.AddressZero,
+            ipfsHash
+          })
+        )
+        return
+      }
+
+      if (`0x${contentHash}` === currentContent) {
+        yield put(
+          fetchENSSuccess({
+            address,
+            name,
+            subdomain,
+            resolver: ENS_RESOLVER_ADDRESS,
+            content: contentHash,
+            ipfsHash,
+            landId: land.id
+          })
+        )
+        return
+      }
+
       yield put(
         fetchENSSuccess({
           address,
           name,
           subdomain,
           resolver: ENS_RESOLVER_ADDRESS,
-          content: landHash,
-          ipfsHash,
-          landId: land.id
+          content: currentContent || ethers.constants.AddressZero,
```

Comment:
The `??` might be semantically a bit better than the or:
```suggestion
          content: currentContent ?? ethers.constants.AddressZero,
```